src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -3951,6 +4206,37 @@ public void CreateTupleTypeSymbol_VisualBasicElements()
         }
 
         [Fact]
+        public void CreateTupleTypeSymbol_ComparingSymbols()
+        {
+            var source = @"
+class C
+{
+    System.ValueTuple<int, int, int, int, int, int, int, (string a, string b)> F;
+}
+";
+
+            var comp = CreateCompilationWithMscorlib(source, references: new[] { ValueTupleRef, SystemRuntimeFacadeRef }, parseOptions: TestOptions.Regular.WithTuplesFeature());
+            var tuple1 = comp.SourceModule.GlobalNamespace.GetMember<NamedTypeSymbol>("C").GetMember<SourceMemberFieldSymbol>("F").Type;
+
+            var intType = comp.GetSpecialType(SpecialType.System_Int32);
+            var stringType = comp.GetSpecialType(SpecialType.System_String);
+
+            var twoStrings = comp.GetWellKnownType(WellKnownType.System_ValueTuple_T2).Construct(stringType, stringType);
+            var twoStringsWithNames = (TypeSymbol)comp.CreateTupleTypeSymbol(twoStrings, ImmutableArray.Create("a", "b"));
+            var tuple2Underlying = comp.GetWellKnownType(WellKnownType.System_ValueTuple_TRest).Construct(intType, intType, intType, intType, intType, intType, intType, twoStringsWithNames);
+            var tuple2 = (TypeSymbol)comp.CreateTupleTypeSymbol(tuple2Underlying);
```

Comment:
I see what you are doing, never mind.

---
In reply to: [63388140](https://github.com/dotnet/roslyn/pull/11160#discussion_r63388140) [](ancestors = 63388140)